cmd/soroban-cli/src/commands/tx/edit.rs

```diff
@@ -0,0 +1,184 @@
+            stdin().read_line(&mut input)?;
+            let input = input.trim();
+            xdr_to_json::<curr::TransactionEnvelope>(input)?
+        };
+
+        let path = tmp_file(&json)?;
+        let editor = get_editor();
+
+        open_editor(&print, &editor, &path)?;
+
+        let contents = fs::read_to_string(&path)?;
+        let xdr = json_to_xdr::<curr::TransactionEnvelope>(&contents)?;
+        fs::remove_file(&path)?;
+
+        println!("{xdr}");
+
+        Ok(())
+    }
+}
+
+struct Editor {
+    cmd: String,
+    source: String,
+    args: Vec<String>,
+}
+
+fn tmp_file(contents: &str) -> Result<PathBuf, Error> {
+    let temp_dir = env::current_dir().unwrap_or(env::temp_dir());
+    let file_name = format!("stellar-json-xdr-{}.json", rand::random::<u64>());
+    let path = temp_dir.join(file_name);
+
+    let mut file = fs::File::create(&path)?;
+    file.write_all(contents.as_bytes())?;
+
+    Ok(path)
+}
+
+fn get_editor() -> Editor {
+    let (source, cmd) = env::var("STELLAR_EDITOR")
+        .map(|val| ("STELLAR_EDITOR", val))
+        .or_else(|_| env::var("EDITOR").map(|val| ("EDITOR", val)))
+        .or_else(|_| env::var("VISUAL").map(|val| ("VISUAL", val)))
+        .unwrap_or_else(|_| ("default", "vim".to_string()));
+
+    let parts: Vec<&str> = cmd.split_whitespace().collect();
+    let cmd = parts[0].to_string();
+    let args = &parts[1..]
+        .iter()
+        .map(|&s| s.to_string())
+        .collect::<Vec<String>>();
+
+    Editor {
+        source: source.to_string(),
+        cmd,
+        args: args.clone(),
+    }
+}
+
+fn open_editor(print: &Print, editor: &Editor, path: &PathBuf) -> Result<(), Error> {
+    print.infoln(format!(
+        "Using `{source}=\"{cmd}\"`",
+        source = editor.source,
+        cmd = editor.cmd,
+    ));
+
+    print.infoln("Opening editor...".to_string());
+
+    let mut binding = process::Command::new(editor.cmd.clone());
+    let command = binding.args(editor.args.clone()).arg(path);
+
+    #[cfg(windows)]
+    let tty = Stdio::null();
+
+    #[cfg(unix)]
+    {
+        let tty = File::open("/dev/tty")?;
+        let tty_out = fs::OpenOptions::new().write(true).open("/dev/tty")?;
+        let tty_err = fs::OpenOptions::new().write(true).open("/dev/tty")?;
+
+        command
+            .stdin(tty)
+            .stdout(tty_out)
+            .stderr(tty_err)
+            .env("TERM", "xterm-256color");
+    }
+
+    let status = command.spawn()?.wait()?;
+
+    if status.success() {
+        Ok(())
+    } else {
+        Err(Error::EditorNonZeroStatus)
+    }
+}
+
+fn xdr_to_json<T>(xdr_string: &str) -> Result<String, Error>
+where
+    T: curr::ReadXdr + serde::Serialize,
+{
+    let xdr_bytes = BASE64.decode(xdr_string)?;
+    let cursor = Cursor::new(xdr_bytes);
+    let mut limit = curr::Limited::new(cursor, curr::Limits::none());
+    let value = T::read_xdr(&mut limit)?;
+    let json = serde_json::to_string_pretty(&value)?;
```

Comment:
Highly recommend using the following function instead:

```
T::from_xdr_base64(...)
```
https://docs.rs/stellar-xdr/latest/stellar_xdr/curr/trait.ReadXdr.html#method.from_xdr_base64

Why:
- It'll ensure the input is fully consumed, that the code here doesn't do and in some situations could be a security concern.
- Don't need to figure out the base64 decoder.

Reply:
Awesome, I was kinda in the dark on what to do, tbh.

Do we have something similar for json->xdr, or is the current approach good enough?

Reply:
I don't think so. What you're doing is 👍🏻